packages/backend-common/src/reading/GithubUrlReader.test.ts

```diff
@@ -271,6 +271,33 @@ describe('GithubUrlReader', () => {
       expect(response.etag).toBe('foo');
       expect(response.lastModifiedAt).toEqual(new Date('2021-01-01T00:00:00Z'));
     });
+
+    it('should override the token if its provided', async () => {
+      expect.assertions(1);
+
+      (mockCredentialsProvider.getCredentials as jest.Mock).mockResolvedValue({
```

Comment:
Optional: If you change the definition of the credentials provider above to

```ts
const mockCredentialsProvider = {
  getCredentials: jest.fn().mockResolvedValue({ headers: {} }),
} satisfies GithubCredentialsProvider;
```

then ALL of these casts all over this file go away